addons/portal/models/portal_mixin.py

```diff
@@ -12,6 +12,14 @@ class PortalMixin(models.AbstractModel):
     portal_url = fields.Char(
         'Portal Access URL', compute='_compute_portal_url',
         help='Customer Portal URL')
+    shared = fields.Boolean('Check shared Document', compute="_compute_shared")
+
+    def _compute_shared(self):
+        # when Access through sudo in portal env.user will be super user
+        user_id = self.env.context.get('uid')
+        partner = self.env['res.users'].browse(user_id).partner_id
+        for record in self.filtered(lambda x: partner in x.message_follower_ids.mapped('partner_id') and  x.create_uid.id != user_id and partner != x.partner_id):
```

Comment:
I am not sure to understand your heuristic here. Could you explain it to me please ?

Reply:
@tde-banana-odoo , main purpose of this compute is to display the 'share'  tag on the portal view.
here current user's partner must be in follower and record must not be created by current user and record must not be own by the current user then we will display 'shared' tag(as previous discussion on the task we can not add a boolean field to check record is shared manually or not). 
and on the portal, several records are accessed through the sudo() so we used context.get('uid') instead of env.user. Thanks